### PR TITLE
implement get_cpu_features function in systemtools

### PR DESCRIPTION
**Pull request by boegel**
This could be used to implement an FFTW easyblock that auto-detects which configure flags can be specified to optimise performance (e.g. `--enable-avx`, `--enable-avx2`, etc.), rather than hardcoding configure flags in the FFTW easyconfigs (which is an issue already, since `--enable-sse2` doesn't work on Linux/POWER for example).

### CODE REVIEW

File: easybuild/tools/systemtools.py [hunk 1]
```diff
@@ -385,6 +385,50 @@ def get_cpu_speed():
...
+        else:
+            _log.debug("%s not found to determine CPU features", PROC_CPUINFO_FP)
+
+    elif os_type == DARWIN:
+        for feature_set in ['extfeatures', 'features', 'leaf7_features']:
+            cmd = "sysctl -n machdep.cpu.%s" % feature_set
+            _log.debug("Trying to determine CPU features on Darwin via cmd '%s'", cmd)
+            out, ec = run_cmd(cmd, force_in_dry_run=True)
+            if ec == 0:
+                # returns clock frequency in cycles/sec, but we want MHz
```
Comment (JensTimmerman) [inline easybuild/tools/systemtools.py]: this doesn't look right at all

File: easybuild/tools/systemtools.py [hunk 2]
```diff
@@ -385,6 +385,50 @@ def get_cpu_speed():
...
+                cpu_feat.extend(out.strip().lower().split())
```
Comment (JensTimmerman) [inline easybuild/tools/systemtools.py]: you probably need to strip out the `xxx:` ?

File: easybuild/tools/systemtools.py (hunk 1 above)
Comment (boegel) [inline easybuild/tools/systemtools.py]: woops, will fix, thanks

**Comment by JensTimmerman** [comment]
lgtm

**Comment by boegel** [comment]
Thanks for the review @JensTimmerman!

**Pull request merged**